### PR TITLE
feat(cli): expose handler hook for injecting SDK options

### PR DESCRIPTION
**Pull request by ronelliott**
### Proposed Changes

* Add a `Hook` extension point to `pkg/handlers` so callers can inject SDK options that depend on the resolved profile without replicating any of the profile resolution or credential validation flow.
* Rename the internal `handlerOptsFunc` to the exported `HandlerOption` so downstream packages can name the option type when composing calls to `handlers.New`.
* Extend `cmd/common.NewHandler` with an optional variadic `...handlers.Hook`. Every existing zero-hook call site keeps working unchanged.

The hook runs after every `HandlerOption` has been applied and before `sdk.New` is called. Hooks compose in registration order and receive a `HookContext` with the fully resolved endpoint, TLS flag, and profile.

### Motivation

A Virtru CLI built on top of otdfctl (`tructl`) currently duplicates `common.NewHandler` wholesale, about 130 lines of profile resolution, flag validation, and credential error handling, purely so it can attach a client-side connect interceptor to the SDK for license status header capture. With this hook that downstream call collapses to:

```go
h := common.NewHandler(c, func(hookCtx handlers.HookContext) []sdk.Option {
    return []sdk.Option{sdk.WithExtraClientOptions(
        connect.WithInterceptors(premium.NewLicenseStatusInterceptor()),
    )}
})
```

and otdfctl stays the single owner of the profile and auth flow.

### Checklist

- [x] I have added or updated unit tests
- [ ] I have added or updated integration tests (if appropriate)
- [ ] I have added or updated documentation

### Testing Instructions

- `cd otdfctl && go test ./... -race`
- New tests live in `pkg/handlers/sdk_test.go` and cover:
  - hook registration order
  - `HookContext` content matches the resolved handler config
  - hook-returned options append to `sdkOpts`
  - `applyHooks` is a no-op when no hooks are registered

## Summary by CodeRabbit

- **New Features**
  - Added an extensibility mechanism for customizing SDK setup through handler hooks.
  - Added support for supplying custom SDK options when creating handlers.
  - Hooks can access resolved endpoint, TLS, and profile settings before SDK initialization.
  - Multiple hooks are supported in registration order, while nil hooks are safely ignored.

- **Tests**
  - Expanded coverage for hook registration, execution order, configuration context, and edge cases.

### CODE REVIEW

File: otdfctl/pkg/handlers/sdk.go [hunk 1]
```diff
@@ -58,20 +78,52 @@ func WithProfile(profile *profiles.OtdfctlProfileStore) handlerOptsFunc {
 	}
 }
 
-func WithSDKOpts(opts ...sdk.Option) handlerOptsFunc {
+func WithSDKOpts(opts ...sdk.Option) HandlerOption {
 	return func(c handlerOpts) handlerOpts {
 		c.sdkOpts = opts
 		return c
 	}
 }
```
[the remark this answers is not in the document]
Comment (ronelliott) [reply otdfctl/pkg/handlers/sdk.go:104]: this is expected behavior, will not update

File: otdfctl/pkg/handlers/sdk.go [hunk 2]
```diff
@@ -29,19 +29,39 @@ type handlerOpts struct {
...
 		c.TLSNoVerify = tlsNoVerify
 		return c
 	}
 }
 
-func WithProfile(profile *profiles.OtdfctlProfileStore) handlerOptsFunc {
+func WithProfile(profile *profiles.OtdfctlProfileStore) HandlerOption {
 	return func(c handlerOpts) handlerOpts {
 		c.profile = profile
 		c.endpoint = profile.GetEndpoint()
```
[the remark this answers is not in the document]
Comment (ronelliott) [reply otdfctl/pkg/handlers/sdk.go:85]: Fixed!

File: otdfctl/pkg/handlers/sdk.go [hunk 3]
```diff
@@ -58,20 +78,52 @@ func WithProfile(profile *profiles.OtdfctlProfileStore) handlerOptsFunc {
...
+// WithHook registers a Hook that runs during handler construction after
+// profile resolution and before the SDK is built. Multiple WithHook calls
+// compose in registration order, and each hook sees the same fully-resolved
+// HookContext.
+func WithHook(hook Hook) HandlerOption {
+	return func(c handlerOpts) handlerOpts {
+		c.hooks = append(c.hooks, hook)
+		return c
+	}
+}
```
[the remark this answers is not in the document]
Comment (ronelliott) [reply otdfctl/pkg/handlers/sdk.go]: Fixed!

**Review by jrschumacher — APPROVED** [review]
